config/environments/prod.rb

```diff
@@ -13,6 +13,11 @@
   # Full error reports are disabled and caching is turned on.
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
+  config.cache_store = :redis_store, ENV['REDISCLOUD_URL'] + '/0/cache',
+                       {
+                         # Needed for redis to evict keys in volatile-lru mode
+                         expires_in: 30.days
```

Comment:
I'm still wondering if redis is the right call for this cache layer. Do we have a good reason to expire the entry after 30 days? Why 30 days? Why not 1 or 365?

Reply:
I chose the default expires to be one month, on the assumption that anything worth seeing will be viewed within a month of precaching.

I realize you advocated for a persistent long term cache.

In any case, I think the best way to decide will be to look at the cache hit rate provided by the metrics in this PR.

Reply:
Question: what are the conditions where this cache needs to be invalidated?

Reply:
I feel like we have no reason to set an arbitrary number. The cache eviction should be either based on memory (ex: Redis should keep up to 1 GB of pre-cached files to avoid memory and performance issues) or would be persistent and rely on well known cache invalidation conditions.

Reply:
By my reading, Redis will act as a LRU cache up to `maxmemory`, according to https://redis.io/topics/lru-cache , so long as we set any `expires_in`.

Reply:
See https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/ParameterGroups.Redis.html#ParameterGroups.Redis.NodeSpecific also. The `maxmemory` currently is 6501171200 or 6.2GB.

Reply:
> what are the conditions where this cache needs to be invalidated?

The conditions are all the cache key param values plus the sample ID plus the expires.

Reply:
Nice to know that we have a `maxmemory`. So why would we need the `expires_in` clause? I see no value on it.

Reply:
Discussed offline. I  will add an `expires_in` in the fetch.